broadinstitute/pearl
# JN-525 site content save

The "save" button now works on the website editor so you can save changes made in the UX.  Hitting save creates a new version of the site content, and then also updates the environment to point to the new version.  As we've talked about before, we eventually may want some 'save as draft' capability, but that's outside the scope of this PR.

In testing, I found that we were missing non-null constraints on siteContent stableId&version.  So that's fixed :)

This also does some drive-by cleanup of environment diffing routes so that the portal environment controller namespace was clearer for the new controller methods needed

TO TEST:
1. go to  https://localhost:3000/ourhealth/studies/ourheart/env/sandbox/siteContent
2. make some changes to the site content
3. hit 'save'
4. confirm the changes are saved as a new version, and you see success notifications telling you the version was created *and* the environment was updated
5. reload the page, confirm your changes persist
6. go to https://localhost:3000/ourhealth/studies/ourheart/publishing
7. confirm you see your latest version as the sandbox version
